src/logger/src/logger.rs

```diff
@@ -1,429 +1,143 @@
-    /// use std::io::Cursor;
-    ///
-    /// use logger::LOGGER;
-    ///
-    /// let mut logs = Cursor::new(vec![0; 15]);
-    ///
-    /// LOGGER.init("Running Firecracker v.x".to_string(), Box::new(logs));
-    /// ```
-    pub fn init(&self, header: String, log_dest: Box<dyn Write + Send>) -> Result<()> {
-        self.init
-            .call_init(|| {
-                let mut g = extract_guard(self.log_buf.lock());
-                *g = log_dest;
-
-                self.try_init_max_level();
-
-                // finish init
-                true
-            })
-            .map_err(LoggerError::Init)?;
-
-        self.write_log(&header);
 
         Ok(())
     }
-
-    /// Handles the common logic of writing regular log messages.
-    ///
-    /// Writes `msg` followed by a newline to the destination, flushing afterwards.
-    fn write_log(&self, msg: &str) {
-        let mut guard;
-        let mut writer: Box<dyn Write> = if self.init.is_initialized() {
-            guard = extract_guard(self.log_buf.lock());
-            Box::new(guard.as_mut())
-        } else {
-            Box::new(stdout())
-        };
-        // Writes `msg` followed by newline and flushes, if either operation returns an error,
-        // increment missed log count.
-        // This approach is preferable over `Result::and` as if `write!` returns  an error it then
-        // does not attempt to flush.
-        if writeln!(writer, "{msg}")
-            .and_then(|_| writer.flush())
-            .is_err()
-        {
-            // No reason to log the error to stderr here, just increment the metric.
-            METRICS.logger.missed_log_count.inc();
-        }
-    }
 }
 
-/// Describes the errors which may occur while handling logging scenarios.
-#[derive(Debug, thiserror::Error)]
-pub enum LoggerError {
-    /// Initialization Error.
-    #[error("Logger initialization failure: {0}")]
-    Init(init::Error),
+#[derive(Debug)]
+pub struct LogFormat {
+    pub show_level: bool,
+    pub show_log_origin: bool,
 }
+#[derive(Debug)]
+pub struct LoggerConfiguration {
+    pub target: Option<std::fs::File>,
+    pub format: LogFormat,
+}
+#[derive(Debug)]
+pub struct Logger(pub Mutex<LoggerConfiguration>);
 
-/// Implements the "Log" trait from the externally used "log" crate.
 impl Log for Logger {
+    // No additional filters to <https://docs.rs/log/latest/log/fn.max_level.html>.
     fn enabled(&self, _metadata: &Metadata) -> bool {
-        // No filtering beyond what the log crate already does based on level.
         true
     }
 
     fn log(&self, record: &Record) {
-        let msg = format!(
-            "{} {} {}",
-            LocalTime::now(),
-            self.create_prefix(record),
-            record.args()
-        );
-        self.write_log(&msg);
+        // Lock the logger.
+        let mut guard = self.0.lock().unwrap();
+
+        // Prints log message
+        {
+            let thread = thread::current().name().unwrap_or("-").to_string();
```

Comment:
```suggestion
            let thread = thread::current();
            let thread = thread.name().unwrap_or("-");
```
one less `to_string()` xP

Reply:
Fixed (I used `thread_name` for the 2nd variable instead of `thread`)